addons/website/static/src/components/fields/fields.js

```diff
@@ -30,11 +30,18 @@ class PageUrlField extends UrlField {
         if (value[0] === "/") {
             value = value.substring(1);
         }
-        // Re-add the leading slash for saving, because url field is required
-        // and thus doesn't accept an empty string.
-        this.props.record.data[this.props.name] = `/${value.trim()}`;
         return value;
     }
+
+    get extraInputFieldParams() {
+        return {
+            ...super.extraInputFieldParams,
+            // Add leading slash for save
+            parse(rawValue) {
+                return rawValue[0] !== "/" ? `/${rawValue}` : rawValue;
+            },
+        };
+    }
```

Comment:
Removing this and its related code. In the end, it has nothing to do with the current fix but it creates a new one apparently :grimacing: 

- Create a new page, and leave "Add in menu" checked
- Discard edit mode that opened
- Go to Site -> Properties
- Change the URL
- Choose to redirect old URL
- Save
=> 
![image](https://github.com/user-attachments/assets/73f00f3d-09ff-4cb8-ba8d-5abec0722a87)

Reply:
The feature itself is probably broken already do:
![image](https://github.com/user-attachments/assets/30c45796-04f5-419e-b497-4a7681cfe805)


Missing "/" ?

Reply:
If issue commit 3 fixes is not there before commit 1, then yes, you decide :wink:

Reply:
> The feature itself is probably broken already do: ![image](https://private-user-images.githubusercontent.com/10338094/372547005-30c45796-04f5-419e-b497-4a7681cfe805.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzNzkwNTcsIm5iZiI6MTcyODM3ODc1NywicGF0aCI6Ii8xMDMzODA5NC8zNzI1NDcwMDUtMzBjNDU3OTYtMDRmNS00MTllLWI0OTctNGE3NjgxY2ZlODA1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA4VDA5MTIzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMxZTM1YjczZmQyZDk1MjU5MGYxZmQxM2NmNmYwMGM2OTlhNDRlOThmZWJlNGM5N2ViMGIwY2JiYWZhMGI5M2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Vs0PAU8qNiglmBlTOc_LucxeDoSSC_gvOvpZNqbz6yw)
> 
> Missing "/" ?

The redirects work even without the leading `/` on the URL to field, so while we don't want that, this is not visible in any meaningful way to the user so I'm just gonna squash them.